src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -5061,9 +5087,13 @@ protected sealed override IAliasSymbol GetAliasInfoCore(SyntaxNode node, Cancell
 
         protected sealed override PreprocessingSymbolInfo GetPreprocessingSymbolInfoCore(SyntaxNode node)
         {
-            return node is IdentifierNameSyntax nameSyntax
-                ? GetPreprocessingSymbolInfo(nameSyntax)
-                : PreprocessingSymbolInfo.None;
+            return node switch
+            {
+                IdentifierNameSyntax nameSyntax => GetPreprocessingSymbolInfo(nameSyntax),
+                DefineDirectiveTriviaSyntax defineSyntax => GetPreprocessingSymbolInfo(defineSyntax, defineSyntax.Name),
+                UndefDirectiveTriviaSyntax undefSyntax => GetPreprocessingSymbolInfo(undefSyntax, undefSyntax.Name),
+                _ => PreprocessingSymbolInfo.None
+            };
```

Comment:
This enables getting preprocessor symbol info directly from `#define` and `#undef` directives. Therefore it is possible to discover references to the (un)defined symbol in these directives.

Reply:
Why do we need that in the compiler?  Aren't these always strings, which you can just get from the syntax?